stackrox/stackrox
# chore(ui): Update test devDependencies in ui

## Description

Update with 3 weeks before 4.5 release.

### cypress

https://github.com/testing-library/cypress-testing-library/releases

* update testing-library/dom version to 10

### dom

https://github.com/testing-library/dom-testing-library/releases

10.1.0 2024-04-29

Features
* Add window events "pagehide" / "pageshow"

10.0.0 2024-04-08

Bug Fixes
* use `defineProperty` on the error object instead of setting the message directly

Features
* Drop support for Node.js 14.x and Node.js 16.x and add support for Node 20.x
* upgrade aria-query to 5.3.0

BREAKING CHANGES
* Minimum supported Node.js version is 18.0
* New version of aria-query changes various roles. Check out the changed tests in 2c57055 to get an overview about what changed.

9.3.4 2024-01-08

Bug Fixes
* Stop calling `waitFor` callback after timeout

### jest-dom

https://github.com/testing-library/jest-dom/releases/tag/v6.0.0

BREAKING CHANGES
* Removes the `extend-expect` script. Users should use
the default import path or one of the new test platform-specific
paths to automatically extend the appropriate "expect" instance.

**Therefore, skip until 4.6A sprint.**

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/setupTests.js#L2

### react

https://github.com/testing-library/react-testing-library/releases/tag/v16.0.0

Features
* Move `@testing-library/dom` and `@types/react-dom` to peer dependencies

BREAKING CHANGES
* `@testing-library/dom` was moved to a peer dependency and needs to be explicitly installed.

https://github.com/testing-library/react-testing-library/releases/tag/v15.0.0

BREAKING CHANGES
* Minimum supported Node.js version is 18.0
* New version 10 of `@testing-library/dom` changes various roles.

https://github.com/testing-library/react-testing-library/releases/tag/v14.3.0

Features
* Add support for React 19 Canary

### user-event

https://github.com/testing-library/user-event/releases

* remove interop and deep DTL imports

### ts-jest

https://github.com/kulshekhar/ts-jest/blob/main/CHANGELOG.md

29.1.4 2024-05-28

Bug Fixes
* fix(transformer): allow transforming of .cts/.mts extensions.
Features
* feat: make cli generate esm config based on `type: "module"`

29.1.3 2024-05-21

Bug Fixes
* add `@jest/transform` as an optional peer dependency
* bring back Node 14 support

Performance Improvements
* remove ts resolved module cache file

29.1.2 2024-01-22

Bug Fixes
* calculated cache key based on supportsStaticESM
* correct error handling in `processAsync`
* use `Config.ProjectConfig`

## Checklist

## Testing Performed

1. `yarn test` in ui/apps/platform and then type `a`
2. CI
